core/src/datayoga_core/blocks/parquet/read/block.py

```diff
@@ -0,0 +1,36 @@
+import logging
+import os
+from abc import ABCMeta
+from typing import Generator, Optional
+
+from datayoga_core.context import Context
+from datayoga_core.producer import Message
+from datayoga_core.producer import Producer as DyProducer
+from fastparquet import ParquetFile
+
+logger = logging.getLogger("dy")
+
+
+class Block(DyProducer, metaclass=ABCMeta):
+
+    def init(self, context: Optional[Context] = None):
+        logger.debug(f"Initializing {self.get_block_name()}")
+        parquet_file = self.properties["file"]
+
+        if os.path.isabs(parquet_file) or context is None:
+            self.file = parquet_file
+        else:
+            self.file = os.path.join(context.properties.get("data_path"), parquet_file)
+
+        logger.debug(f"file: {self.file}")
+
+    def produce(self) -> Generator[Message, None, None]:
+        logger.debug("Reading parquet")
+
+        pf = ParquetFile(self.file)
+
+        count = 0
+        for df in pf.iter_row_groups():
```

Comment:
I think the batch_size should be configurable.

Reply:
`fastparquet` does not have this capability. It can read only by row_group witch is actually bucket.

Reply:
https://fastparquet.readthedocs.io/en/latest/api.html